### PR TITLE
Allow equals sign in an environment variable value

### LINKED ISSUE #4142: Unable to pass environment variables containing `=` to `az container create`
### Description
**Outline the issue here:**

An example scenario is passing a connection string to a container as environment variable.

To further illustrate, the command below:

```
az container create -g SomeGroup --name something --image some/image -e 'EVENT_HUBS_CONNECTION_STRING=Endpoint=sb://foo.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=bar='
```

Will result into the following error:

```
Incorrectly formatted enviroment settings. Argmuent values should be in the format a=b c=d
```

### PR DESCRIPTION
**Pull request by vjrantal**
Fixes #4142.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ x ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ x ] Each command and parameter has a meaningful description.
- [ x ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by tjprescott** [comment]
Fixes #4084 as well.

File: src/command_modules/azure-cli-container/HISTORY.rst [hunk 1]
```diff
@@ -1,5 +1,10 @@
 .. :changelog:
 
+unreleased
+----------
+
+* Allow equals sign in an environment variable value.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-container/HISTORY.rst]: Recommend this be more specific:  "container create: Fixes issue where equals sign was not allowed inside an environment variable.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py [hunk 2]
```diff
@@ -19,12 +19,13 @@
 
 def environment_variables_format(value):
     """Space separated values in 'key=value' format."""
-    try:
-        env_name, env_value = value.split('=')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py]: I would recommend something more concise, like so:
```Python
try:
  env_name, env_value = value.split('=', 1)
```

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/tests/test_container_commands.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ def test_container_create(self, resource_group, resource_group_location):
         cpu = 1
         memory = 1
         command = '"/bin/sh -c \'while true; do echo hello; sleep 20; done\'"'
-        env = 'KEY1=VALUE1 KEY2=VALUE2'
+        env = 'KEY1=VALUE1 KEY2=FOO=BAR='
```
Comment (tjprescott) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/tests/test_container_commands.py]: 👍 for the test :smile:

**Comment by vjrantal** [comment]
@tjprescott Thanks for the review! Amended the PR content so that it address all of your comments.

**Comment by tjprescott** [comment]
@annatisch please consider whether this fix is appropriate for `azure batch` (I can't see why it wouldn't be)

**Pull request merged**